pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/kubernetes/KubernetesRuntime.java

```diff
@@ -973,8 +982,18 @@ private V1PodSpec getPodSpec(List<String> instanceCommand, Function.Resources re
     V1Container getFunctionContainer(List<String> instanceCommand, Function.Resources resource) {
         final V1Container container = new V1Container().name(PULSARFUNCTIONS_CONTAINER_NAME);
 
-        // set up the container images
-        container.setImage(pulsarDockerImageName);
+        Function.FunctionDetails.Runtime runtime = instanceConfig.getFunctionDetails().getRuntime();
+
+        if (runtime == Function.FunctionDetails.Runtime.JAVA){
```

Comment:
I don't think this is a backwards compatible change.  Existing users would still be using "pulsarDockerImageName" and only that config will be set.  "pulsarDockerImageName" should still be the default config used for image name and the new configs can override it if set.